tests/compile-fail/transmute.rs

```diff
@@ -0,0 +1,38 @@
+#![feature(core)]
+#![feature(plugin)]
+#![plugin(clippy)]
+
+extern crate core;
+
+use std::mem::transmute as my_transmute;
+use std::vec::Vec as MyVec;
+
+fn my_vec() -> MyVec<i32> {
+    vec![]
+}
+
```

Comment:
Could you add some tests on generics? One test which tests for `transmute<T,T>` (which might trigger the lint), and one for `transmute<T,U>` (which should not). It's okay if the lint doesn't warn on `<T,T>` -- in that case leave a comment on the test to that effect.